blockmanager.go

```diff
@@ -1343,11 +1343,37 @@ func (b *blockManager) detectBadPeers(headers map[string]*wire.MsgCFHeaders,
 
 	log.Warnf("Detected cfheader mismatch at height=%v!!!", targetHeight)
 
-	// Get the block header for this height, along with the block as well.
+	// Get the block header for this height.
 	header, err := b.server.BlockHeaders.FetchHeaderByHeight(targetHeight)
 	if err != nil {
 		return nil, err
 	}
+
+	// Fetch filters from the peers in question.
+	// TODO(halseth): query only peers from headers map.
+	filtersFromPeers := b.fetchFilterFromAllPeers(
+		targetHeight, header.BlockHash(), fType,
+	)
+
+	var badPeers []string
+	for peer, _ := range headers {
+		_, ok := filtersFromPeers[peer]
+
+		// If a peer did not respond, ban it immediately.
+		if !ok {
```

Comment:
This might be an issue for nodes with higher latency which would end up running into timeouts when fetching the filter.

Reply:
This is to ensure peers cannot avoid the filter check by just not responding to our query.

Reply:
Shouldn't be too big of an issue with the increased timeout introduced in #158.

Reply:
We'll definitely want to make this timeout dynamic in the future. We can also use traffic shaping utilities locally to emulate high latency links as well.